src/index.js

```diff
@@ -18,7 +18,28 @@ const DEFAULT_ALLOW_HEADERS = [
 
 const DEFAULT_MAX_AGE_SECONDS = 60 * 60 * 24 // 24 hours
 
-const cors = (options = {}) => handler => (req, res, ...restArgs) => {
+function isString (s) {
+  return typeof s === 'string' || s instanceof String
+}
+
+function isOriginAllowed (origin, allowedOrigin) {
+  if (Array.isArray(allowedOrigin)) {
+    for (var i = 0; i < allowedOrigin.length; ++i) {
+      if (isOriginAllowed(origin, allowedOrigin[i])) {
+        return true
+      }
+    }
+    return false
+  } else if (isString(allowedOrigin)) {
+    return origin === allowedOrigin
+  } else if (allowedOrigin instanceof RegExp) {
+    return allowedOrigin.test(origin)
+  } else {
+    return !!allowedOrigin
```

Comment:
What is this final else statement's purpose?

Reply:
if we set `options.origin` to `true` every domain should be valid.
for every other value that is not handled above CORS should be disabled.

Reply:
I think we should remove using boolean in `origin`. `true` already exists with `*`, and `false` already exists by removing the library. 

Although, does the callback function rely on this?

Reply:
i disagree. i think it's more elegant to switch this behavior on and off by just using boolean instead of removing the whole library, especially when you want to change this behavior in a programmatic way. what should we do?

the callback function does not rely on this btw.

Reply:
It’s true that it’s more elegant, but do we expect users to be switching cors on and off regularly? 

When the function PR is added and merged, couldn’t the user make a function that always returns true or false to cover this case? 

Another concern is that if `allowedOrigin` is not an array, string, or regex but is an object or something else it will act as true. That’s obviously user error but still a concern.

Reply:
you are probably right. it could be done with the function PR. should i remove the last if statement? what should happen if `allowedOrigin` is an object or boolean?

Reply:
Let's change it to `return false` to be on the safe side.